mathoxide-lib/src/array.rs

```diff
@@ -85,4 +107,24 @@ mod test {
         let formatted = array.to_string();
         assert_eq!(formatted.split('\n').count(), array.shape()[0]);
     }
+
+    #[test]
+    fn reshape_2d_to_1d() {
+        let array = Array::<ThreadSafeStorage<u32>, ContiguousView>::zeros(&[2, 3]);
+        // Temporary until getters for Array are available
+        println!("{}", array);
+        let row_vec = array.reshape([6usize]);
+        println!("{}", row_vec);
+    }
+
+    #[test]
+    fn reshape_2d_to_1d_inferred() {
+        let array = Array::<ThreadSafeStorage<u32>, ContiguousView>::zeros(&[2, 3]);
+        // Temporary until getters for Array are available
+        println!("{}", array);
+        let row_vec = array.reshape([-1isize]);
+        println!("{}", row_vec);
+        let col_vec = array.reshape([-1isize, 1]);
+        println!("{}", col_vec);
```

Comment:
maybe instead of the `println`s we could at least check that the view inside the array is translating properly using the new shape?

Reply:
I am not planning to merge until we add a proper getter and write a proper test anyway so this is not an issue